components/script/dom/htmlmediaelement.rs

```diff
@@ -1552,6 +1569,70 @@ impl HTMLMediaElement {
             .start(0)
             .unwrap_or_else(|_| self.playback_position.get())
     }
+
+    fn render_controls(&self) {
+        let element = self.htmlelement.upcast::<Element>();
+        if self.ready_state.get() < ReadyState::HaveMetadata || element.is_shadow_host() {
+            // Bail out if we have no metadata yet or
+            // if we are already showing the controls.
+            return;
+        }
+        if let Ok(shadow_root) = element.attach_shadow(IsUserAgentWidget::Yes) {
```

Comment:
You should just `unwrap()` the result, right? No exception should be thrown if you're a UA widget.